samples/nrf_desktop/CMakeLists.txt

```diff
@@ -25,13 +34,23 @@ macro(set_conf_file)
 	"configuration/nrfconnect.conf"
 	"configuration/zephyr.conf"
 	)
+
+  if (${CMAKE_BUILD_TYPE} STREQUAL ZRelease)
+  else()
+    list(APPEND CONF_FILE "configuration/zephyr_dbg.conf")
+  endif()
+
+  if (${CMAKE_BUILD_TYPE} STREQUAL ZDebugWithShell)
+    list(APPEND CONF_FILE "configuration/app_shell.conf")
```

Comment:
I would name `app_shell.conf` `zephyr_shell.conf` for consistency with `zephyr_dbg.conf`. They are both additional configurations.

Reply:
For both comments: I though about naming it like that the first place but named it app as it contains CONFIG_DESKTOP_xxx config and so affects application configuration. In the current app config files we have configuration specific to zephyr. We could separate it and create zephyr_board_name files but it would cause files to multiply. At the moment it is easy enough to follow I guess and the number of files is limited.

Reply:
Is it possible/desirable to come up with a naming scheme where the CONF_FILE can be deduced from the CMAKE_BUILD_TYPE?

Reply:
Having such a scheme sound like a good idea. Although I like the split into base config, `dbg` and `shell`, the proposed new way will have less conflicts and probably will be easier to maintain.

Reply:
This sounds reasonable in the short-term.

In the long-term I hope that https://github.com/zephyrproject-rtos/zephyr/issues/8181 will be resolved.

Then one could drop the .conf fragments and instead express in Kconfig sources that there exists a choice of configurations and that each configuration depends on / requires a different set of options.